app.py

```diff
@@ -103,23 +103,22 @@ def main():
             # Handle the case of no configuration file resulting in a 404 response code
 
             # Update default config with those provided
-            try:
-                with open(".pep8speaks.yml", "r") as stream:
+            with open(".pep8speaks.yml", "r") as stream:
+                try:
                     new_config = yaml.load(stream)
-                    config["ignore"] = new_config["ignore"]
-                    for act in ["opened", "updated"]:
-                        for pos in ["header", "footer"]:
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                            config["message"][act][pos] = new_config[
-                                "message"][act][pos].replace("{name}", author)
-                    config["scanner"]["diff_only"] = new_config["scanner"]["diff_only"]
-            except:  # Bad yml file
-                pass
+                    # overriding the default configuration with the one specified
+                    config = new_config
+
+                except yaml.YAMLError as e:  # Bad YAML file
+                    pass
+
+            # personalising the messages
+            if "message" in config:
+                for act in config["message"].keys():
+                    # can be either "opened" or "updated"
+                    for pos in act.keys():
+                        # can be either "header" or "footer"
+                        config["message"][act][pos].replace("{name}", author)
```

Comment:
I can't remember what the hell was I doing repeating the same line four times. This is nice!